File: longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
        HashSet<Integer> s = new HashSet<>();
        
        for (int i : nums) { s.add(i); }
        
        int lcs = 0;
        
        for (Integer j : s) {
            
            if (s.contains(j - 1) == false) { 
                
                int current = j + 1;
                
                while (s.contains(current) == true) { ++current; }
                
                lcs = Math.max(lcs, current - j);
            }
        }
        
        return lcs;
    }
}

File: longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        std::unordered_set<int> s(nums.begin(), nums.end());
        
        int lcs = 0;
        
        for (int i : s) {
            
            if (s.find(i - 1) == s.end()) { 
                
                int current = i + 1;
                
                // std::unordered_set::erase returns 1 if a key value was found and erased
                while (s.erase(current) == 1) { ++current; }
                
                lcs = std::max(lcs, current - i);
            }
        }
        
        return lcs;
    }
};